src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs

```diff
@@ -1,78 +1,108 @@
 
+using System;
 using System.Collections.Generic;
 using System.CommandLine;
-using System.CommandLine.Invocation;
 using System.CommandLine.Parsing;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.Tools.MSBuild;
 using Microsoft.DotNet.Tools.New;
+using Microsoft.DotNet.Tools.Restore;
+using Microsoft.TemplateEngine.Cli;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.TemplatePackage;
+using Microsoft.TemplateEngine.Edge;
+using System.Linq;
 
 namespace Microsoft.DotNet.Cli
 {
     internal static class NewCommandParser
     {
         public static readonly string DocsLink = "https://aka.ms/dotnet-new";
+        public const string CommandName = "new";
+        private const string HostIdentifier = "dotnetcli";
 
-        public static readonly Argument Argument = new Argument<IEnumerable<string>>() { Arity = ArgumentArity.ZeroOrMore };
+        private static readonly Option<bool> _disableSdkTemplates = new Option<bool>("--debug:disable-sdk-templates", () => false, "If present, prevents templates bundled in the SDK from being presented").Hide();
 
-        public static readonly Option ListOption = new Option<bool>(new string[] { "-l", "--list" });
+        internal static readonly System.CommandLine.Command Command = GetCommand();
 
-        public static readonly Option NameOption = new Option<string>(new string[] { "-n", "--name" });
-
-        public static readonly Option OutputOption = new Option<string>(new string[] { "-o", "--output" });
-
-        public static readonly Option InstallOption = new Option<bool>(new string[] { "-i", "--install" });
-
-        public static readonly Option UninstallOption = new Option<bool>(new string[] { "-u", "--uninstall" });
-
-        public static readonly Option InteractiveOption = new Option<bool>("--interactive");
-        
-        public static readonly Option NuGetSourceOption = new Option<string>("--nuget-source");
-        
-        public static readonly Option TypeOption = new Option<string>("--type");
-        
-        public static readonly Option DryRunOption = new Option<bool>("--dry-run");
-        
-        public static readonly Option ForceOption = new Option<bool>("--force");
-        
-        public static readonly Option LanguageOption = new Option<string>(new string[] { "-lang", "--language" });
-
-        public static readonly Option UpdateCheckOption = new Option<bool>("--update-check");
-
-        public static readonly Option UpdateApplyOption = new Option<bool>("--update-apply");
-
-        public static readonly Option ColumnsOption = new Option<bool>("--columns");
-
-        private static readonly Command Command = ConstructCommand();
-
-        public static Command GetCommand()
+        internal static readonly Option LanguageOption = Command.Options.SingleOrDefault(o => o.Name == "-lang");
```

Comment:
this logic is currently incorrect - @vlada-shubina we need a way to retrieve the Language Option correctly in order to support the Telemetry Filter.

Reply:
alternative - remove logging the language at this stage in telemetry. The Template Engine already collects the language of template invocations, so this data point isn't very useful.